src/services/apimService.ts

```diff
@@ -28,7 +29,7 @@ export class ApimService extends BaseService {
     }
 
     if (!this.apimConfig.name) {
-      this.apimConfig.name = `${this.config.provider.prefix}-${this.config.provider.region}-${this.config.provider.stage}-apim`
+      this.apimConfig.name = `${this.config.provider.prefix}-${Utils.createShortAzureRegionName(this.config.provider.region)}-${this.config.provider.stage}-apim`
```

Comment:
We should call into Apim Resource.getResourceName instead. This is what is used when resource is initially generated.

Reply:
Updated to use the exiting function.